syntax = "proto3";

// Interface of APM clients to be used for shared testing.
service APMClient {
  rpc StartSpan(StartSpanArgs) returns (StartSpanReturn) {}
  rpc FinishSpan(FinishSpanArgs) returns (FinishSpanReturn) {}
  rpc FlushSpans(FlushSpansArgs) returns (FlushSpansReturn) {}
}

message StartSpanArgs {
  string name = 1;
  optional string service = 2;
  optional uint64 parent_id = 3;
  // TODO
}
message StartSpanReturn {
  uint64 span_id = 1;
  uint64 trace_id = 2;
}


message FinishSpanArgs {
  uint64 id = 1;
}
message FinishSpanReturn {}

message FlushSpansArgs {}
message FlushSpansReturn {}
